module-system/module-implementations/sov-blob-storage/src/capabilities.rs

```diff
@@ -0,0 +1,88 @@
+use sov_modules_api::capabilities::{BlobRefOrOwned, BlobSelector};
+use sov_modules_api::{Context, Spec};
+use sov_rollup_interface::da::BlobReaderTrait;
+use sov_state::WorkingSet;
+use tracing::info;
+
+use crate::BlobStorage;
+
+impl<C: Context> BlobSelector for BlobStorage<C> {
+    type Context = C;
+
+    fn get_blobs_for_this_slot<'a, I, B>(
+        &self,
+        current_blobs: I,
+        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
+    ) -> anyhow::Result<Vec<BlobRefOrOwned<'a, B>>>
+    where
+        B: BlobReaderTrait,
+        I: IntoIterator<Item = &'a mut B>,
+    {
+        // TODO: Chain-state module: https://github.com/Sovereign-Labs/sovereign-sdk/pull/598/
+        let current_slot: u64 = self.get_current_slot_number(working_set);
+        let past_deferred: Vec<B> = current_slot
+            .checked_sub(self.get_deferred_slots_count(working_set))
+            .map(|pull_from_slot| self.take_blobs_for_block_number(pull_from_slot, working_set))
+            .unwrap_or_default();
+        let preferred_sequencer = self.get_preferred_sequencer(working_set);
+
+        let preferred_sequencer = if let Some(sequencer) = preferred_sequencer {
+            sequencer
+        } else {
+            return Ok(past_deferred
+                .into_iter()
+                .map(Into::into)
+                .chain(current_blobs.into_iter().map(Into::into))
```

Comment:
This line means that we'll process all the blobs from both the previous block and the current block at once if the sequencer exits. Are we sure that's safe?

Reply:
That's a good point.

So no blob from unregistered sequencer is going to be actually executed, because `apply_blob_hook` from sequencer registry will figure them out.

But overall gas usage might be 2x normal block size. And that might be a problem. 

But what would be solution?

The code can chain only half of the current blobs and save the rest for the next block, and repeat it until there's only 4 blobs saved (for instance).

Reply:
As discussed on the call, let's move this issue out of the scope of this PR.

Tracking issue: https://github.com/Sovereign-Labs/sovereign-sdk/issues/654